pubsub/subject.go

```diff
@@ -34,13 +31,13 @@ type Subject interface {
 // subject is the internal implementation of Subject
 type subject struct {
 	observers map[string]*observer
-	mut       sync.RWMutex
```

Comment:
why we changed to mutex?

Reply:
`RWMutex` is redundant as there is no use for readers lock